src/Modules/ModulesGraded.jl

```diff
@@ -2688,3 +2688,43 @@ function twist(M::ModuleFP{T}, d::IntegerUnion) where {T<:MPolyDecRingElem}
   @assert is_z_graded(R)
   return twist(M, grading_group(R)([d]))
 end
+
+# TODO: implement further base changes. But for this we need more functionality 
+# for potential change of grading groups. See the open pr #2677.
+function change_base_ring(
```

Comment:
This actually became quite relevant for me in this PR: `change_base_ring` for graded modules and translating the gradings. For special cases where I know the gradings don't change, you can customize this. But in general, I was forced to introduce an internal helper routine below. Maybe this can one day be used in a front-end method as a workhorse for the case of preserved gradings.